test/visualTest.ts

```diff
@@ -123,56 +125,68 @@ describe("DualKpi", () => {
+                }
+            };
 
-                    if (value.displayName) {
-                        expect(value.displayNameKey).toBeDefined();
-                    }
+            visualBuilder.updateRenderTimeout(dataView, () => {
+                expect(visualBuilder.pathAreas.length).toBeGreaterThan(2);
+                expect(visualBuilder.axisTicks.length).toBe(4);
+                done();
+            });
+        });
 
-                    if (typeof value === "object") {
-                        objectsChecker(value);
-                    }
+        it("top chart should be rendered", (done) => {
+            dataView.metadata.objects = {
+                dualKpiProperties: {
+                    topChartShow: true,
+                    bottomChartShow: false,
                 }
             };
 
-            objectsChecker(jsonData);
+            visualBuilder.updateRenderTimeout(dataView, () => {
+                expect(visualBuilder.pathAreas.length).toBeGreaterThan(1);
+                expect(visualBuilder.axisTicks.length).toBe(2);
+                done();
+            });
         });
-    });
 
-    describe("DOM tests", () => {
-        it("svg element created", () => expect(visualBuilder.mainElement[0]).toBeInDOM());
+        it("bottom chart should be rendered", (done) => {
+            dataView.metadata.objects = {
+                dualKpiProperties: {
+                    topChartShow: false,
+                    bottomChartShow: true,
+                }
+            };
 
-        it("update", (done) => {
             visualBuilder.updateRenderTimeout(dataView, () => {
-                expect(visualBuilder.chartGroup.children("path.area"))
-                    .toBeInDOM();
-                expect(visualBuilder.chartGroup.children("g.axis").children("g.tick").length)
-                    .toBe(4);
+                expect(visualBuilder.pathAreas.length).toBeGreaterThan(1);
+                expect(visualBuilder.axisTicks.length).toBe(2);
                 done();
             });
         });
 
+
         it("update with null values", (done) => {
-            dataView.categorical.categories[0].values[0] = null;
-            dataView.categorical.values[0].values[0] = null;
+            dataView.categorical!.categories![0].values[0] = null as unknown as PrimitiveValue;
+            dataView.categorical!.values![0].values[0] = null as unknown as PrimitiveValue;
 
             visualBuilder.updateRenderTimeout(dataView, () => {
-                expect(visualBuilder.chartGroup.children("path.area")).toBeInDOM();
+                expect(visualBuilder.pathAreas.length).toBeGreaterThan(0);
                 done();
             }, DefaultWaitForRender);
         });
 
         it("changed data with high contrast on", () => {
-            let dataColorTop: string = getRandomHexColor(),
-                textColorTop: string = getRandomHexColor(),
+            const dataColorTop: string = "#01B8AA",
```

Comment:
why not to use random colors with getRandomHexColor() as it was before?

Reply:
I remember changing it after talking with Aidar. He said that the test shouldn't be random and it's easier to test when the tests are predictable.

Personally, I prefer to change back to the random colors, this way we'll test all colors instead of just one.
What do you think?

Reply:
I agree, so we can test random colors behavior as well. As a compromise, we can add additional similar test case with randomization as it was before and a strict one as Aidar suggested.

Reply:
Done.